### PR TITLE
Enable libp2p peer discovery and test

### PR DESCRIPTION
**Pull request by LeoHChen**
## Issue

#373 

#### Test/Run Logs

the original test/deploy.sh works fine.  I just added a new parameter "-P" to enable libp2p peer discovery test.
Also, for harmony.go, I added a parameter -libp2p_pd for testing as well.

## TODO

- [ ] integrate 2nd stage peer discovery
- [ ] integrate leader and all pubkeys support

### CODE REVIEW

File: test/deploy.sh [hunk 1]
```diff
@@ -131,8 +135,23 @@ $DRYRUN $ROOT/bin/beacon -numShards $SHARDS > $log_folder/beacon.log 2>&1 | tee
 sleep 1 #waiting for beaconchain
 MA=$(grep "Beacon Chain Started" $log_folder/beacon.log | awk -F: ' { print $2 } ')
 
+echo "launching boot node ..."
+$DRYRUN $ROOT/bin/bootnode > $log_folder/bootnode.log 2>&1 | tee -a $LOG_FILE &
+sleep 1
+BN_MA=$(grep "BN_MA" $log_folder/bootnode.log | awk -F\= ' { print $2 } ')
```
Comment (harmony-ek) [inline test/deploy.sh]: This type of information valuable to outside users should be exposed via a more suitable means than just log extraction.  Log parsing shouldn't be part of a deployment pipeline.  I'd say write a one-liner file and then read from the script.  Like a PID file.

File: test/deploy.sh [hunk 2]
```diff
@@ -131,8 +135,23 @@ $DRYRUN $ROOT/bin/beacon -numShards $SHARDS > $log_folder/beacon.log 2>&1 | tee
 sleep 1 #waiting for beaconchain
 MA=$(grep "Beacon Chain Started" $log_folder/beacon.log | awk -F: ' { print $2 } ')
```
Comment (harmony-ek) [inline test/deploy.sh]: MA → BC_MA maybe?

File: test/deploy.sh [hunk 3]
```diff
@@ -131,8 +135,23 @@ $DRYRUN $ROOT/bin/beacon -numShards $SHARDS > $log_folder/beacon.log 2>&1 | tee
...
 
+echo "launching boot node ..."
+$DRYRUN $ROOT/bin/bootnode > $log_folder/bootnode.log 2>&1 | tee -a $LOG_FILE &
+sleep 1
+BN_MA=$(grep "BN_MA" $log_folder/bootnode.log | awk -F\= ' { print $2 } ')
+
+HMY_OPT=
+HMY_OPT2=
```
Comment (harmony-ek) [inline test/deploy.sh]: Maybe use `HMY_OPT+=...` and do away with this extra variable?

Comment (LeoHChen) [reply test/deploy.sh]: HMY_OPT2 is not applicable to txgen, that's why I have to use two variables here.

File: test/deploy.sh (hunk 2 above)
Comment (LeoHChen) [reply test/deploy.sh]: sure, I can change that.

File: test/deploy.sh (hunk 1 above)
Comment (LeoHChen) [reply test/deploy.sh]: agree. this is an ugly hack. I have opened an issue for a better design.
https://github.com/harmony-one/harmony/issues/396

**Pull request merged**